ethereum/graphqlrpc/src/main/java/tech/pegasys/pantheon/ethereum/graphqlrpc/internal/pojoadapter/PendingStateAdapter.java

```diff
@@ -0,0 +1,128 @@
+import tech.pegasys.pantheon.ethereum.eth.transactions.PendingTransactions;
+import tech.pegasys.pantheon.ethereum.graphqlrpc.GraphQLDataFetcherContext;
+import tech.pegasys.pantheon.ethereum.graphqlrpc.internal.BlockchainQuery;
+import tech.pegasys.pantheon.ethereum.graphqlrpc.internal.TransactionWithMetadata;
+import tech.pegasys.pantheon.ethereum.mainnet.ProtocolSchedule;
+import tech.pegasys.pantheon.ethereum.transaction.CallParameter;
+import tech.pegasys.pantheon.ethereum.transaction.TransactionSimulator;
+import tech.pegasys.pantheon.ethereum.transaction.TransactionSimulatorResult;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import graphql.schema.DataFetchingEnvironment;
+
+@SuppressWarnings("unused") // reflected by GraphQL
+public class PendingStateAdapter extends AdapterBase {
+
+  private final PendingTransactions pendingTransactions;
+
+  public PendingStateAdapter(final PendingTransactions pendingTransactions) {
+    this.pendingTransactions = pendingTransactions;
+  }
+
+  public Integer getTransactionCount() {
+    return pendingTransactions.size();
+  }
+
+  public List<TransactionAdapter> getTransactions() {
+    return pendingTransactions.getTransactionInfo().stream()
+        .map(PendingTransactions.TransactionInfo::getTransaction)
+        .map(TransactionWithMetadata::new)
+        .map(TransactionAdapter::new)
+        .collect(Collectors.toList());
+  }
+
+  // until the miner can expose the current "proposed block" we have no
+  // speculative environment, so estimate against latest.
+  public Optional<AccountAdapter> getAccount(
+      final DataFetchingEnvironment dataFetchingEnvironment) {
+    final BlockchainQuery blockchainQuery =
+        ((GraphQLDataFetcherContext) dataFetchingEnvironment.getContext()).getBlockchainQuery();
+    final Address addr = dataFetchingEnvironment.getArgument("address");
+    final Long bn = dataFetchingEnvironment.getArgument("blockNumber");
+    final long latestBn = blockchainQuery.latestBlock().get().getHeader().getNumber();
+    final Optional<WorldState> ows = blockchainQuery.getWorldState(latestBn);
+    return ows.flatMap(ws -> Optional.ofNullable(ws.get(addr))).map(AccountAdapter::new);
+  }
+
+  // until the miner can expose the current "proposed block" we have no
+  // speculative environment, so estimate against latest.
+  public Optional<Long> getEstimateGas(final DataFetchingEnvironment environment) {
+    final Optional<CallResult> result = executeCall(environment);
+    return result.map(CallResult::getGasUsed);
+  }
+
+  // until the miner can expose the current "proposed block" we have no
+  // speculative environment, so estimate against latest.
+  public Optional<CallResult> getCall(final DataFetchingEnvironment environment) {
+    return executeCall(environment);
```

Comment:
There are two methods specified in the EIP-1767, one that returns the entire call result and one that wants us to estimate the gas.  Executing the call and getting the gasUsed is the easiest way to estimate, so they are deduped by calling executeCall (which is not exposed) and returning what we need.  It could be simplified by making getCall be executeCall.